src/helpers/feed-helpers.ts

```diff
@@ -0,0 +1,111 @@
+import axios from 'axios'
+
+// eslint-disable-next-line import/no-cycle
+import { FeedsApiClientV20210630 } from '../api-clients/feeds-api-client-v20210630'
+
+/**
+ * Sleep helper
+ * @param secs
+ * @returns
+ */
+function sleep(secs: number) {
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve()
+    }, secs * 1000)
+  })
+}
+
+/**
+ * Submit a feed to Amazon Selling Partner
+ * @param feedsApiClient
+ * @param feedType
+ * @param content
+ * @param amazonMarketplaceId: 'ATVPDKIKX0DER' (US)
+ * @param contentType: 'application/xml
+ * @returns
+ */
+export async function submitFeedHelper(
+  feedsApiClient: FeedsApiClientV20210630,
+  feedType: string,
+  content: string,
+  amazonMarketplaceId = 'ATVPDKIKX0DER',
+  contentType = 'application/xml',
+): Promise<string> {
+  // create the feed document
+  const feedDocumentResult = await feedsApiClient.createFeedDocument({
+    body: {
+      contentType,
+    },
+  })
+
+  // upload the file
+  await axios.request({
+    url: feedDocumentResult.data.url,
+    method: 'PUT',
+    headers: {
+      'Content-Type': contentType,
+    },
+    data: content,
+  })
+
+  // submit feed
+  const createFeedResponse = await feedsApiClient.createFeed({
+    body: {
+      feedType,
+      marketplaceIds: [amazonMarketplaceId],
+      inputFeedDocumentId: feedDocumentResult.data.feedDocumentId,
+    },
+  })
+
+  return createFeedResponse.data.feedId
+}
+
+/**
+ * Given a feedClient and feedId, will loop 15 times x4 seconds delay to fetch the feed results
+ * We will parse the XML only to look for errors, which should help the user decide if they need to parse further
+ * @param feedsApiClient
+ * @param feedId
+ * @returns { hasErrors: boolean, xmlResponse: string }
+ */
+export async function getFeedResult(
+  feedsApiClient: FeedsApiClientV20210630,
+  feedId: string,
+): Promise<{ hasErrors: boolean; xmlResponse: string }> {
+  let resultFeedDocumentId = ''
+  let feedStatus = ''
+  let attempts = 0
+  while (feedStatus !== 'DONE') {
+    // eslint-disable-next-line no-await-in-loop
+    await sleep(4)
```

Comment:
Should attempts and sleep time be configurable? Is it guaranteed to never take longer than 60 seconds?

Reply:
Good call. I'll just make these configurable with defaults. It's not always going to be those.